openmdao/solvers/nonlinear/newton.py

```diff
@@ -190,6 +192,25 @@ def _iter_execute(self):
         # Disable local fd
         approx_status = system._owns_approx_jac
         system._owns_approx_jac = False
+        
+        # This cycles through the solvers of the subsystems once at the start of the problem
+        if self.options['initial_sub_solve'] == True:
```

Comment:
I agree, and think instead we use three options that work together:

solve_subsystems -- Turns on sub solves
max_sub_solves -- Max number of iterations of sub solve per execution
initial_sub_solve_only -- Only do sub solves the first time it executes, then no more.

then we can use the existing code, modify the `do_subsolve` logic a little bit, and just save the exec count